rewrite-maven/src/main/java/org/openrewrite/maven/AddManagedDependency.java

```diff
@@ -0,0 +1,167 @@
+            example = "org.apache.logging.log4j.*")
+    private String onlyIfUsing;
+
+    @Option(displayName = "Classifier",
+            description = "A Maven classifier to add. Most commonly used to select shaded or test variants of a library",
+            example = "test",
+            required = false)
+    @Nullable
+    private String classifier;
+
+    @Option(displayName = "Add to the root pom",
+            description = "Add to the root pom where root is the eldest parent of the pom within the source set.",
+            example = "true",
+            required = false)
+    private Boolean addToRootPom;
+
+    @Override
+    public Validated validate() {
+        Validated validated = super.validate();
+        //noinspection ConstantConditions
+        if (version != null) {
+            validated = validated.or(Semver.validate(version, versionPattern));
+        }
+        return validated;
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Add Managed Maven Dependency";
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getApplicableTest() {
+        return new UsesType<>(onlyIfUsing);
+    }
+
+    @Override
+    protected List<SourceFile> visit(List<SourceFile> before, ExecutionContext ctx) {
+        Map<JavaProject, Set<String>> sourceSetsByProject = new HashMap<>();
+        final AtomicReference<SourceFile> rootPom = new AtomicReference<>();
+        for (SourceFile source : before) {
+            source.getMarkers().findFirst(JavaProject.class).ifPresent(javaProject ->
+                    source.getMarkers().findFirst(JavaSourceSet.class).ifPresent(sourceSet -> {
+                        if (source != new UsesType<>(onlyIfUsing).visit(source, ctx)) {
+                            sourceSetsByProject.computeIfAbsent(javaProject, v -> new HashSet<>()).add(sourceSet.getName());
+                        }
+                    }));
+            source.getMarkers().findFirst(MavenResolutionResult.class).ifPresent(mavenResolutionResult -> {
+                if (rootPom.get() == null && mavenResolutionResult.getParent() == null) {
+                    rootPom.set(source);
+                }
+            });
+        }
+
+        if (sourceSetsByProject.isEmpty()) {
+            return before;
+        }
+
+        return ListUtils.map(before, s -> s.getMarkers().findFirst(JavaProject.class)
+                .map(javaProject -> (Tree) new MavenVisitor<ExecutionContext>() {
+                    @Override
+                    public Xml visitDocument(Xml.Document document, ExecutionContext executionContext) {
+                        Xml maven = super.visitDocument(document, executionContext);
+
+                        if ("test".equals(scope) && !sourceSetsByProject.get(javaProject).contains("test")) {
+                            return maven;
+                        }
+
+                        if (!addToRootPom || (rootPom.get() != null && rootPom.get().equals(document))) {
```

Comment:
`!addToRootPom` results in a NPE if `addToRootPom` is not set (field is not required). This should be replaced with e.g. `!Boolean.TRUE.equals(addToRootPom) `

Reply:
Thanks @thomaszub :)